src/gardenlinux/github/__main__.py

```diff
@@ -0,0 +1,836 @@
+        # Skip if no publishing metadata found
+        if published_image_metadata is None:
+            continue
+
+        platform = metadata["platform"]
+        arch = metadata["architecture"]
+
+        # Determine variant from flavor name
+        flavor_name = metadata["s3_key"].split("/")[-1]
+        variant = get_variant_from_flavor(flavor_name)
+
+        if variant not in grouped_data:
+            grouped_data[variant] = {}
+        if platform not in grouped_data[variant]:
+            grouped_data[variant][platform] = {}
+        if arch not in grouped_data[variant][platform]:
+            grouped_data[variant][platform][arch] = []
+
+        grouped_data[variant][platform][arch].append(metadata)
+
+    # Generate both table and old format
+    output = "## Published Images\n\n"
+
+    # Add table format (collapsed by default)
+    output += "<details>\n<summary>📊 Table View</summary>\n\n"
+    output += generate_table_format(grouped_data)
+    output += "\n</details>\n\n"
+
+    # Add old format (collapsed by default)
+    output += "<details>\n<summary>📝 Detailed View</summary>\n\n"
+    output += generate_detailed_format(grouped_data)
+    output += "\n</details>\n\n"
+
+    return output
+
+
+def generate_table_format(grouped_data):
+    """
+    Generate the table format with collapsible region details
+    """
+    output = "| Variant | Platform | Architecture | Flavor | Regions & Image IDs | Download Links |\n"
+    output += "|---------|----------|--------------|--------|---------------------|----------------|\n"
+
+    for variant in VARIANT_ORDER:
+        if variant not in grouped_data:
+            continue
+
+        for platform in sorted(grouped_data[variant].keys()):
+            platform_display = get_platform_display_name(platform)
+
+            for arch in sorted(grouped_data[variant][platform].keys()):
+                # Process all metadata for this variant/platform/architecture
+                for metadata in grouped_data[variant][platform][arch]:
+                    data = get_platform_release_note_data(metadata, platform)
+                    if data is None:
+                        continue
+
+                    # Generate collapsible details for regions
+                    details_content = generate_region_details(data, platform)
+                    summary_text = generate_summary_text(data, platform)
+
+                    # Generate download links
+                    download_links = generate_download_links(data["flavor"], platform)
+
+                    # Use shorter names for table display
+                    variant_display = VARIANT_TABLE_NAMES[variant]
+                    output += f"| {variant_display} | {platform_display} | {arch} | `{data['flavor']}` | <details><summary>{summary_text}</summary><br>{details_content}</details> | <details><summary>Download</summary><br>{download_links}</details> |\n"
+
+    return output
```

Comment:
This looks like markdown. Function name should reflect that. Not sure if I like the mixing of HTML and markdown.